symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/BdkConfigLoader.java

```diff
@@ -0,0 +1,156 @@
+package com.symphony.bdk.core.config;
+
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import com.symphony.bdk.core.config.legacy.model.LegacySymConfig;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+@Slf4j
+public class BdkConfigLoader {
+
+    private static final JsonMapper JSON_MAPPER = new JsonMapper();
+    private static final YAMLMapper YAML_MAPPER = new YAMLMapper();
+
+    /**
+     * Load BdkConfig from a file path
+     *
+     * @param configPath Path of the config file
+     *
+     * @return Symphony Bot Configuration
+     */
+    public static BdkConfig loadFromFile(String configPath) {
+        try {
+            return JSON_MAPPER.readValue(new File(configPath), BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in JSON format");
+        }
+
+        try {
+            return YAML_MAPPER.readValue(new File(configPath), BdkConfig.class);
+        } catch (IOException e) {
+            log.debug("Config is not in Yaml format");
+        }
+
+        log.error("Config is not in a valid format");
+        return null;
+    }
+
+    /**
+     * Load BdkConfig from an InputStream
+     *
+     * @param inputStream InputStream
+     *
+     * @return Symphony Bot Configuration
+     */
+    public static BdkConfig loadFromInputStream(InputStream inputStream) {
+        try {
```

Comment:
this could be factorized with the code above (calling loadFromInputStream from loadFromFile

also perhaps we could try to detect the format based on the extension, using the exceptions feels a bit ugly

Reply:
I also have an idea to detect the format by using regular expression but I'm not sure if it's a good way or not. I searched from internet and they all suggested using exceptions.